muspinsim_input/build_file.py

```diff
@@ -0,0 +1,381 @@
+                            "field", [parse_field(entry) for entry in values]
+                        ),
+                        # either scalar or single function
+                        "times": lambda values: build_block(
+                            "time",
+                            [
+                                " ".join(split_into_args(entry["time"], 1))
+                                for entry in values
+                            ],
+                        ),
+                        # either scalar or single function
+                        "temperatures": lambda values: build_block(
+                            "temperature",
+                            [
+                                " ".join(
+                                    split_into_args(entry["temperature"], 1)
+                                )
+                                for entry in values
+                            ],
+                        ),
+                        "x_axis": lambda value: build_block("x_axis", [value]),
+                        "y_axis": lambda value: build_block("y_axis", [value]),
+                        "average_axes": lambda values: build_block(
+                            "average_axes",
+                            [v.strip() for v in values.split(",")],
+                        ),
+                        "experiment_preset": lambda value: build_block(
+                            "experiment", [value]
+                        ),
+                        "orientations": lambda values: build_block(
+                            "orientation {0}".format(euler_convention),
+                            [parse_orientation(entry) for entry in values],
+                        ),
+                        "interactions": lambda values: "".join(
+                            [parse_interactions(entry) for entry in values]
+                        ),
+                        "polarizations": lambda values: build_block(
+                            "polarization",
+                            [parse_polarization(entry) for entry in values],
+                        ),
+                        "fitting": lambda value: build_block(
+                            "fitting_data", ['load("fitting_data.dat")']
+                        ),
+                        "fitting_method": lambda value: build_block(
+                            "fitting_method", [value]
+                        ),
+                        "fitting_variables": lambda values: build_block(
+                            "fitting_variables",
+                            [
+                                parse_fitting_variables(entry)
+                                for entry in values
+                            ],
+                        ),
+                        "fitting_tolerance": lambda value: build_block(
+                            "fitting_tolerance",
+                            [str(value)],
+                        ),
```

Comment:
Sorry for the ambiguity - I meant the dictionary would be best placed outside the `main` function altogether. Where it is, when I start reading/skimming past it I forget what the code was trying to do by the time I get to the end 😅

Reply:
So move it out into another function?

Reply:
Since the dictionary is constant I'm thinking more of just using a variable:
```
keyword_functions = {
    "spins": lambda values: build_block(...),
    ...
}
                        

def main():
    ...
    keyword_func = keyword_functions.get(keyword, None)
    if keyword_func:
        file_contents.append(keyword_func(val))
    ...